idaes/power_generation/unit_models/drum1D.py

```diff
@@ -0,0 +1,1171 @@
+                         doc="Outside Heat Transfer Coefficient")
+
+        # Insulation free convection heat transfer coefficient
+        self.h_free_conv = Var(self.flowsheet().config.time, initialize=1,
+                               doc="Insulation Free Convection"
+                               "Heat Transfer Coefficient")
+
+        # Ra number of free convection
+        self.N_Ra_root6 = Var(self.flowsheet().config.time, initialize=80,
+                              doc="1/6 Power of Ra"
+                              "Number of Free Convection of Air")
+
+        # Nu number  of free convection
+        self.N_Nu = Var(self.flowsheet().config.time, initialize=1,
+                        doc="Nu Number of Free Convection of Air")
+
+        # Define the continuous domains for model
+        self.r = ContinuousSet(bounds=(self.drum_ri, self.drum_ro))
+
+        # Temperature across wall thickness
+        self.drum_wall_temperature = Var(self.flowsheet().config.time,
+                                         self.r,
+                                         bounds=(280, 800),
+                                         initialize=550)
+
+        # Declare derivatives in the model
+        if self.config.dynamic is True:
+            self.dTdt = DerivativeVar(self.drum_wall_temperature,
+                                      wrt=self.flowsheet().config.time)
+        self.dTdr = DerivativeVar(self.drum_wall_temperature,
+                                  wrt=self.r)
+        self.d2Tdr2 = DerivativeVar(self.drum_wall_temperature,
+                                    wrt=(self.r, self.r))
+
+        discretizer = TransformationFactory('dae.finite_difference')
+        discretizer.apply_to(self,
+                             nfe=self.config.finite_elements,
+                             wrt=self.r,
+                             scheme='CENTRAL')
+
+        # Add performance variables
+        self.level = Var(
+                self.flowsheet().config.time,
+                initialize=1.0,
+                doc='Water Level from the Bottom of the Drum')
+
+        # Velocity of fluid inside downcomer pipe
+        self.velocity_downcomer = Var(
+                self.flowsheet().config.time,
+                initialize=10.0,
+                doc='Liquid Water Velocity at the Top of Downcomer')
+
+        # Pressure change due to contraction
+        self.deltaP_contraction = Var(
+                self.flowsheet().config.time,
+                initialize=-1.0,
+                doc='Pressure Change due to Contraction')
+
+        # Pressure change due to gravity
+        self.deltaP_gravity = Var(
+                self.flowsheet().config.time,
+                initialize=1.0,
+                doc='Pressure Change due to Gravity')
+
+        # Thermal diffusivity of drum
+        @self.Expression(doc="Thermal Diffusivity of Drum Material")
+        def diff_therm_metal(b):
+            return b.therm_cond_metal / (b.dens_metal * b.cp_metal)
```

Comment:
Whilst these are useful for simplifying things later, do we really need to attach them to the model? I.e. why not do:

```
diff_therm_metal = b.therm_cond_metal / (b.dens_metal * b.cp_metal)
```

This achieves the same goal later on, but does not attach the expression to the model, saving us some memory overhead.

Notably, you do this later for the thermal stress calculations.